packages/asset-service/src/service/AssetService.ts

```diff
@@ -21,22 +22,33 @@ export class AssetService {
     return this.assets
   }
 
-  async description(assetId: AssetId): Promise<DescriptionData> {
-    const descriptions: Record<string, string> = assetsDescriptions
-    const description = descriptions[assetId]
-
+  async description(assetId: AssetId, locale = 'en'): Promise<DescriptionData> {
+    const localeDescriptions = assetsDescriptions[locale]
     // Return overridden asset description if it exists and add isTrusted for description links
-    if (description) return { description, isTrusted: true }
+    if (assetsDescriptions.en[assetId]) {
+      const polyglot = new Polyglot({
+        phrases: localeDescriptions,
+        allowMissing: true,
+        onMissingKey: (key) => assetsDescriptions.en[key] // fallback to English overriden description, which should always be added as a base translation
+      })
+      const overriddenDescription = polyglot.t(assetId) as string
+
+      return { description: overriddenDescription, isTrusted: true }
+    }
 
     try {
-      type CoinData = { description: { en: string } }
+      type CoinData = { description: { [locale: string]: string } }
 
       const url = adapters.makeCoingeckoAssetUrl(assetId)
       if (!url) throw new Error()
 
       const { data } = await axios.get<CoinData>(url)
 
-      return { description: data?.description?.en ?? '' }
+      const description = data?.description[locale]?.length
+        ? data?.description?.[locale]
+        : data?.description.en ?? ''
```

Comment:
alternative
`data?.description?.[locale] || data?.description?.en ?? ''`

Reply:
Not sure about this one, it was originally on the attempted bounty PR https://github.com/shapeshift/lib/pull/843#discussion_r908809250 but I always prefer to use the nullish coalescing operator for consistency and keep short circuits for explicitly truthy/falsy values

Reply:
we can also do a check on `if (!data?.description) return { description: ''}` to save reduce the optional chains if that would make it more readable

Reply:
first `||` will fall back if undefined or empty string in the case no locale found so we can be sure to check for en description in both undefined or empty string case

Reply:
That sounds cleaner! Added in b54505b98e9ddfa2e55de5817fb6b3f464463459 and removed not-needed optional chaining. It looks like `data?.description[locale]` should always be defined (but possibly empty) so we can safely remove one more optional chaining

Reply:
> first `||` will fall back if undefined or empty string in the case no locale found so we can be sure to check for en description in both undefined or empty string case

`Boolean('')` is truthy so using `||` would just move the check back somewhere else haha

Reply:
> > first `||` will fall back if undefined or empty string in the case no locale found so we can be sure to check for en description in both undefined or empty string case
> 
> `Boolean('')` is truthy so using `||` would just move the check back somewhere else haha

yeah to our fallback en.

`data.description[locale]` being either undefined or `''` would fall back to `data.descriptions.en` with the `||`